src/main/java/org/apache/commons/csv/CSVFormat.java

```diff
@@ -2097,6 +2097,30 @@ public CSVParser parse(final Reader reader) throws IOException {
         return new CSVParser(reader, this);
     }
 
+    /**
+     * Parses the specified content.
+     *
+     * <p>
+     * This method provides a way to parse CSV data from an input stream, starting at a specified character offset and record number,
+     * using a specified encoding. It returns a {@link CSVParser} that can be used to iterate over the parsed {@link CSVRecord}s.
+     * </p>
+     *
+     * <p>
+     * For additional parsing options, see the various static parse methods available on {@link CSVParser}.
+     * </p>
+     *
+     * @param reader the input stream
+     * @param characterOffset the character offset to start parsing from
+     * @param recordNumber the initial record number to start counting from
+     * @param encoding the character encoding of the input stream
+     * @return a parser over a stream of {@link CSVRecord}s.
+     * @throws IOException If an I/O error occurs
+     * @throws CSVException Thrown on invalid input.
+     */
+    public CSVParser parse(final Reader reader, final long characterOffset, final long recordNumber, String encoding) throws IOException {
```

Comment:
No new public constructors please. You can augment the builder and add a private constructor.

All new and protected elements should have a Java `@since 1.13.0`.

Reply:
I noticed that there is only builder class in CSVFormat, there are no builder class in CSVParser, would you like me to add one?
There are CSVParserBuilder() from com.opencsv.CSVParser, that is another package.

Reply:
Uh? See https://github.com/apache/commons-csv/blob/96427fc72fe5873fd0bd23cac147eef302bf5b75/src/main/java/org/apache/commons/csv/CSVParser.java#L151

Reply:
It seems like I did not rebase on the master branch. Thanks.

Reply:
This section is not resolved.